### ISSUE TITLE
sagemaker: Support new-style personal CfnSpace

### ISSUE THREAD

**Issue by athewsey** [aws/aws-cdk#28985]
### Describe the feature

CDK should support personal SageMaker Spaces (as [introduced for re:Invent 2023](https://aws.amazon.com/blogs/machine-learning/boost-productivity-on-amazon-sagemaker-studio-introducing-jupyterlab-spaces-and-generative-ai-tools/)) - not just the "classic" shared spaces currently supported by [CfnSpace](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_sagemaker.CfnSpace.html).

The current CfnSpace construct only seems to support some "classic" shared space types, not the [full range](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sagemaker-space-spacesettings.html#cfn-sagemaker-space-spacesettings-apptype).

### Use Case

As described on the [blog](https://aws.amazon.com/blogs/machine-learning/boost-productivity-on-amazon-sagemaker-studio-introducing-jupyterlab-spaces-and-generative-ai-tools/), Personal Spaces are now targeted as the default experience of SageMaker Studio, with the old 'Apps' platform still available in "Studio Classic".

We can already create SageMaker Domains and User Profiles within CDK, and would like to be able to create personal (JupyterLab, Code Editor, etc) Spaces for them in line with this change.

### Proposed Solution

Update the model behind [CfnSpace](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_sagemaker.CfnSpace.html) to align with the new broader capabilities of [CloudFormation AWS::SageMaker::Space](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sagemaker-space.html#cfn-sagemaker-space-spacename).

### Other Information

If anybody has workarounds for creating a personal `JupyterLab` space until the L1 `CfnSpace` construct is updated, I'd appreciate it!

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.126.0

### Environment details (OS name and version, etc.)

macOS Sonoma

**Comment by pahud** [comment]
I believe CfnSpace is actually AWS::SageMaker::Space resource.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sagemaker-space.html

I am not sure if Personal Spaces are supported in CFN. Did you find any document about that?

**Comment by athewsey** [comment]
Yes, can confirm Personal Spaces are supported in CFn because I've since been able to deploy one by making liberal use of `CfnSpace.add_override()`, something like:

```python
personal_space = sagemaker_cdk.CfnSpace(
    self,
    "PersonalSpace",
    domain_id=studio_domain.domain_id,
    space_name=studio_user.name + "-space",
)
personal_space.add_override("Properties.SpaceSettings.AppType", "JupyterLab")
personal_space.add_override(
    "Properties.SpaceSettings.SpaceStorageSettings.EbsStorageSettings.EbsVolumeSizeInGb", 10
)
personal_space.add_override(
    "Properties.OwnershipSettings.OwnerUserProfileName", studio_user.name
)
personal_space.add_override("Properties.SpaceSharingSettings.SharingType", "Private")
```

**Comment by kaizencc** [comment]
Facts. I see `SpaceSettingsProperty` typed as such in the latest CDK:

```ts
  export interface SpaceSettingsProperty {
    /**
     * The JupyterServer app settings.
     *
     * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sagemaker-space-spacesettings.html#cfn-sagemaker-space-spacesettings-jupyterserverappsettings
     */
    readonly jupyterServerAppSettings?: cdk.IResolvable | CfnSpace.JupyterServerAppSettingsProperty;

    /**
     * The KernelGateway app settings.
     *
     * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sagemaker-space-spacesettings.html#cfn-sagemaker-space-spacesettings-kernelgatewayappsettings
     */
    readonly kernelGatewayAppSettings?: cdk.IResolvable | CfnSpace.KernelGatewayAppSettingsProperty;
  }

```

This looks like it was a recent update to the CloudFormation schema as a snapshot of the schema from ~2 weeks ago did not have these new properties. That means that we will naturally get the updated spec and release soon, either this week or next.

Look out for our next PR that looks like this: https://github.com/aws/aws-cdk/pull/28878. The comment diff should include the new properties you are looking for.

**Comment by aemada-aws** [comment]
All `CfnSpace` CFN properties are now available in CDK, closing as completed. If I missed something, please recreate the issue on the latest CDK version.
